### DISCUSSION TITLE
Week 3 FreeCodeCamp Java Likes, Dislikes, and Suggestions

### DISCUSSION THREAD

**Discussion by gabebott** [USU-ITLS-x265-23-summer/class-repo#87] [Ideas]
I liked how comprehensive the curriculum was. I felt like it covered many different concepts and aspects of JavaScript very well. I think the platform also did a great job of encouraging hands-on problem solving in the different tasks. I was able to test my skills and learn from mistakes. It felt like there was instant feedback and aid throughout the learning process.

 I do think it could have gone into more depth on certain topics. I think it could have used more examples of real world applications on certain topics throughout the course. I think this could have helped in figuring out what exactly the end goal was for a few of these coding scenarios.

**Comment by metasean** [comment]
> I think it could have used more examples of real world applications on certain topics throughout the course.

I completely agree.  I suspect one of the reasons they didn't is that there is a _lot_ of material to work through, and using it in projects would make it even more time to work through than it already is.

**Comment by metasean** [comment]
Also, this is all JavaScript.  
Java is a completely different language.  
The naming is [somewhat intentionally confusing](https://dev.to/dboatengx/history-of-javascript-how-it-all-began-92a#javascript-vs-java). 

>Java is to JavaScript what Car is to Carpet.  
Chris Heilmann

**Comment by kassyoveson** [comment]
"Java is to JavaScript what Car is to Carpet." -Chris Heilmann. That is super funny and very true. I looked at the link that Sean added. It is very interesting to read how JavaScript got its name and the history of it.
